core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/MakeAsyncHttpRequestStage.java

```diff
@@ -119,7 +122,11 @@ public MakeAsyncHttpRequestStage(TransformingAsyncResponseHandler<OutputT> respo
                                                                 .fullDuplex(isFullDuplex(context.executionAttributes()))
                                                                 .build();
 
+        Timer timer = MetricUtils.timer(context.attemptMetricRegistry(), SdkDefaultMetric.HttpRequestRoundTripLatency);
```

Comment:
Can we extract the timer code to a separate method, say `startTimer()`?

Reply:
What's the benefit here? I can change MetricUtils.timer method to start and return the timer but not really sure how much that is saving...

Reply:
To me, the benefit is to isolate metrics logic and to keep `executeHttpRequest` method smaller, simpler and just doing one thing (though it's already doing multiple things right now)